src/fileimpl.cpp

```diff
@@ -334,6 +334,17 @@ makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsi
         l = p;
       else
       {
+        // We have found a entry with the correct title.
+        // But nothing guaranties this entry title is unique.
+        // We may have entries before "p" with the same title.
+        while (p>l) {
+          d = getDirentByTitle(title_index_t(--p));
+          if (direntCompareTitle(ns, title, *d)!=0) {
+            // We went one time to far
+            p++;
+            break;
+          }
+        }
         log_debug("article found after " << itcount << " iterations in file \"" << getFilename() << "\" at index " << p);
         return { true, title_index_t(p) };
       }
```

Comment:
I think that this `else` branch can be dropped completely and the binary search logic changed in favor of finding the lower bound of the query value. Then the return statements following the binary search loop will return the correct result.

Reply:
Done is a separated commit. I think it is a important change enough to not have a fixup commit here.